validation-test/stdlib/Collection/LazyMapCollection.swift

```diff
@@ -0,0 +1,145 @@
+// workaround is needed to link all required libraries in case we compile
+// StdlibUnittest with -sil-serialize-all.
+import SwiftPrivate
+#if _runtime(_ObjC)
+import ObjectiveC
+#endif
+
+var CollectionTests = TestSuite("Collection")
+
+// Test collection as sequence using value types as elements.
+CollectionTests.addSequenceTests(
+  makeSequence: { (elements: [OpaqueValue<Int>]) in
+    elements.lazy.map(identity)
+  },
+  wrapValue: identity,
+  extractValue: identity,
+  makeSequenceOfEquatable: { (elements: [MinimalEquatableValue]) in
+    elements.lazy.map(identityEq)
+  },
+  wrapValueIntoEquatable: identityEq,
+  extractValueFromEquatable: identityEq
+)
+
+// Test collection as sequence using reference types as elements.
+CollectionTests.addSequenceTests(
+  makeSequence: { (elements: [LifetimeTracked]) in
+    elements.lazy.map { $0 }
+  },
+  wrapValue: { (element: OpaqueValue<Int>) in
+    LifetimeTracked(element.value, identity: element.identity)
+  },
+  extractValue: { (element: LifetimeTracked) in
+    OpaqueValue(element.value, identity: element.identity)
+  },
+  makeSequenceOfEquatable: { (elements: [LifetimeTracked]) in
+    elements.lazy.map { $0 }
+  },
+  wrapValueIntoEquatable: { (element: MinimalEquatableValue) in
+    LifetimeTracked(element.value, identity: element.identity)
+  },
+  extractValueFromEquatable: { (element: LifetimeTracked) in
+    MinimalEquatableValue(element.value, identity: element.identity)
+  }
+)
+
+// Test collection as bidirectional collection using value types as elements.
+CollectionTests.addBidirectionalCollectionTests(
+  makeCollection: { (elements: [OpaqueValue<Int>]) in
+    elements.lazy.map(identity)
+  },
+  wrapValue: identity,
+  extractValue: identity,
+  makeCollectionOfEquatable: { (elements: [MinimalEquatableValue]) in
+    elements.lazy.map(identityEq)
+  },
+  wrapValueIntoEquatable: identityEq,
+  extractValueFromEquatable: identityEq
+)
+
+// Test collection as bidirectional collection using reference types as elements.
+CollectionTests.addBidirectionalCollectionTests(
+  makeCollection: { (elements: [LifetimeTracked]) in
+    elements.lazy.map { $0 }
+  },
+  wrapValue: { (element: OpaqueValue<Int>) in
+    LifetimeTracked(element.value, identity: element.identity)
+  },
+  extractValue: { (element: LifetimeTracked) in
+    OpaqueValue(element.value, identity: element.identity)
+  },
+  makeCollectionOfEquatable: { (elements: [LifetimeTracked]) in
+    elements.lazy.map { $0 }
+  },
+  wrapValueIntoEquatable: { (element: MinimalEquatableValue) in
+    LifetimeTracked(element.value, identity: element.identity)
+  },
+  extractValueFromEquatable: { (element: LifetimeTracked) in
+    MinimalEquatableValue(element.value, identity: element.identity)
+  }
+)
+
+// Test collection as random access collection using value types as elements.
+CollectionTests.addRandomAccessCollectionTests (
+  makeCollection: { (elements: [OpaqueValue<Int>]) in
+    elements.lazy.map(identity)
```

Comment:
You should wrap elements into a minimal collection of appropriate kind.  This will check for more issues.